pkg/kubelet/dockertools/manager.go

```diff
@@ -261,23 +271,53 @@ func (self *DockerManager) GetPodStatus(pod *api.Pod) (*api.PodStatus, error) {
 		}
 		dockerContainerName := dockerName.ContainerName
 		c, found := expectedContainers[dockerContainerName]
-		terminationMessagePath := ""
 		if !found {
-			// TODO(dchen1107): should figure out why not continue here
-			// continue
-		} else {
-			terminationMessagePath = c.TerminationMessagePath
+			continue
 		}
-		// We assume docker return us a list of containers in time order
+		terminationMessagePath := c.TerminationMessagePath
+		if containerDone.Has(dockerContainerName) {
+			continue
+		}
+		// We assume docker returns a list of containers sorted reversedly by
+		// time.
 		if containerStatus, found := statuses[dockerContainerName]; found {
```

Comment:
I know this is slightly unrelated, but I'm worried this is gonna bite us in the future. containerStatus is actually a copy of what's inside statuses which is why we have to re-add it. Do you think that's clear in the flow today? I worry if someone adds a continue somewhere and silently we don't update the status after we edited it. A possible fix is to have the value be pointers instead of objects.